### DISCUSSION TITLE
wire:navigate and event (duplication of component - sql)

### DISCUSSION THREAD

**Discussion by yooplait** [livewire/livewire#6200] [1. Help]
Good evening,

Imagine a component like this...
 
 ```
class FileBrowsing extends Component
{
    use WithPagination;

    private IAssetService $assetService;
    #[Locked] public ?array $search = null;

    public function boot(IAssetService $assetService)
    {
        $this->assetService = $assetService;
    }

    #[On('search-user')]
    public function search(array $data)
    {
        $this->search = $data;
    }

    public function render()
    {
        $assets = $this->assetService->findBy($this->search)->paginate(); $assets->load('thumbnail');
        return view('livewire.file-browsing')->with('files', $assets);
    }
}
 ```
 Imagine a button that dispatches an event like this, elsewhere in the design
 
```
@click="$dispatch('search-user', { 'data' : {{json_encode(['content_types' => [\App\Enums\MimeTypeCategory::Image]])}} })"
```

So when the user clicks, the event is intercepted and the render runs (and therefore an sql query), which is normal.
 
Imagine a wire:navigate somewhere else

 ```
<a href="{{ route('files.show', $folder->uuid) }}" wire:navigate>
     {{ $folder->name }}
</a> 
```

<img width="892" alt="image" src="https://github.com/livewire/livewire/assets/25279695/2ab5723b-105c-428b-99fa-b38a9850d821">

The problem now is that if the user clicks the button that fires the "search-user" event, the sql query fires twice, because there are now two FileBrowsing components.

And each time the user uses the wire:navigate, a new FileBrowsing is created and the sql queries are duplicated...

<img width="855" alt="image" src="https://github.com/livewire/livewire/assets/25279695/6ed534f9-b493-4848-9d85-988cac1247ef">

Do you know a solution?

I hope I was clear enough, being French :)

Thank you so much !

**Comment by yooplait** [comment]
Hi, 

@persist('browsing') <livewire:file-browsing :root="$root" /> @endpersist

This can be a solution but... during the wire-navigate, the folder should change, which is obviously not the case with a wire-navigate.

**Comment by joshhanley** [comment]
@yooplait thanks for reporting! I think I understand the issue, it seems the events are doubling up when being sent back to the server. I think it might be related to this issue - see my comment regarding old events in request https://github.com/livewire/livewire/discussions/6095#discussioncomment-6668366

**Comment by joshhanley** [comment]
PR #6095 has been fixed and merged. So this will be in the next release. 🙂 can you test it then and let us know if there are any issues? Thanks!

**Reply by yooplait** [reply]
It's working now :) Thank you so much 🙂

**Comment by yooplait** [comment]
Hi,

Suddenly back... Weird.

Thank you for your help

**Comment by yooplait** [comment]
Fixed again. For some reason, i was with "livewire/livewire": "3.0.0-beta.7" because of a new package.
No problem with "livewire/livewire": "3.0.0-beta.8",
